src/tools/cephfs/cephfs_shell.py

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/env python3
+# coding=utf-8
+
+import argparse
+import os
+import sys
+import cmd
+import cephfs as libcephfs
+import shlex
+import shutil
+from cephfsfile import CephFSFile
+import traceback
+import colorama
+
+
+cephfs =  None
+
+def setup_cephfs():
+    """ Mouting a cephfs """
+    global cephfs
+    cephfs = libcephfs.LibCephFS(conffile='/home/admin1/Documents/ceph/build/ceph.conf')
+    cephfs.mount()
+
+def get_file(file_name):
+    if not isinstance(file_name, bytes):
+        file_name = bytes(file_name, encoding="ascii")
+    try :
+       return cephfs.open(file_name, 'r', 0o755) 
+    except : #handling file exception
+       print( "\nFile not found\n") 
+       return -1
+
+def get_data(file_name):
+    fd = get_file(file_name)
+    if fd == -1 :
+        return bytes("",encoding="ascii")
+    return cephfs.read(fd, 0, cephfs.fstat(fd).st_size) 
+
+def write_file(file_name, data):
+    if not isinstance(file_name, bytes):
+      file_name = bytes(file_name, encoding="ascii")
+    fd = cephfs.open(file_name, 'w', 0o755)   
+    cephfs.write(fd, bytes(data, encoding="ascii"), 0)
+
+def list_items(dir_name = ''):
+    if dir_name == '':
+        d = cephfs.opendir(cephfs.getcwd())
+    else :
+        d = cephfs.opendir(dir_name)
+    dent = cephfs.readdir(d)
+    items = []
+    while dent:
+        if dent.d_name not in [b".", b".."]:
+            items.append(dent)
+        dent = cephfs.readdir(d)
+    cephfs.closedir(d)
+    return items
+
+def print_details(file_name, flag):
+    fd = cephfs.fstat(get_file(file_name))
+    print(fd.st_mode, fd.st_size, fd.st_uid, fd.st_gid, fd.st_mtime, colorama.Fore.BLUE * flag + file_name + colorama.Style.RESET_ALL * flag, sep = '\t')
+
+def copy_from_local(local_path, remote_path):
+    data = ""
+    for i in open(local_path):
```

Comment:
Where do you read from the local file? 'open' just returns the file object, right?
And I think you'd want open in binary mode.

You'd want to read in byte chunks (size 4MB? not sure about the size) from local file and write that
byte chunk into the CephFS file. Finally close the fd of the local file and the CephFS file.

Note that you can't read the entire contents of a very large file at once. What if the file size is much bigger than the machine's memory?